rapids-cmake/cpm/cuco.cmake

```diff
@@ -0,0 +1,86 @@
+#=============================================================================
+# Copyright (c) 2022, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#=============================================================================
+include_guard(GLOBAL)
+
+#[=======================================================================[.rst:
+rapids_cpm_cuco
+---------------------
+
+.. versionadded:: v22.08.00
+
+Allow projects to find or build `cuCollections` via `CPM` with built-in
+tracking of these dependencies for correct export support.
+
+Uses the version of cuCollections :ref:`specified in the version file <cpm_versions>` for consistency
+across all RAPIDS projects.
+
+.. code-block:: cmake
+
+  rapids_cpm_cuco( [BUILD_EXPORT_SET <export-name>]
+                   [INSTALL_EXPORT_SET <export-name>]
+                   )
+
+``BUILD_EXPORT_SET``
+  Record that a :cmake:command:`CPMFindPackage(cuco)` call needs to occur as part of
+  our build directory export set.
+
+``INSTALL_EXPORT_SET``
+  Record a :cmake:command:`find_dependency(cuco) <cmake:module:CMakeFindDependencyMacro>` call needs to occur as part of
+  our install directory export set.
+
+Result Targets
+^^^^^^^^^^^^^^
+  cuco::cuco target will be created
+
+#]=======================================================================]
+function(rapids_cpm_cuco)
+  list(APPEND CMAKE_MESSAGE_CONTEXT "rapids.cpm.cuco")
+
+  set(options)
+  set(one_value INSTALL_EXPORT_SET)
+  set(multi_value)
+  cmake_parse_arguments(CUCO "${options}" "${one_value}" "${multi_value}" ${ARGN})
```

Comment:
Please make the prefix `_RAPIDS` to match the changes in https://github.com/rapidsai/rapids-cmake/pull/203

Reply:
I just saw that PR and commented on that choice there, then I saw this comment here. I will wait on that discussion getting finished up before changing this.